pkg/cmd/roachtest/roachtestutil/utils.go

```diff
@@ -65,6 +67,49 @@ func (e *EveryN) ShouldLog() bool {
 	return e.ShouldProcess(timeutil.Now())
 }
 
+// GetWorkloadHistogramArgsString creates a histogram flag string based on the roachtest to pass to workload binary
+// This is used to make use of t.ExportOpenmetrics() method and create appropriate exporter
+func GetWorkloadHistogramArgsString(
+	t test.Test, c cluster.Cluster, labels map[string]string,
+) string {
+	var histogramArgs string
+	if t.ExportOpenmetrics() {
+		// Add openmetrics related labels and arguments
+		histogramArgs = fmt.Sprintf(" --histogram-export-format='openmetrics' --histograms=%s/%s --openmetrics-labels='%s'",
+			t.PerfArtifactsDir(), GetBenchmarkMetricsFileName(t), clusterstats.GetOpenmetricsLabelString(t, c, labels))
+	} else {
+		// Since default is json, no need to add --histogram-export-format flag in this case and also the labels
+		histogramArgs = fmt.Sprintf(" --histograms=%s/%s", GetBenchmarkMetricsFileName(t), t.PerfArtifactsDir())
+	}
+
+	return histogramArgs
+}
+
+// GetBenchmarkMetricsFileName returns the file name to store the benchmark output
+func GetBenchmarkMetricsFileName(t test.Test) string {
+	if t.ExportOpenmetrics() {
+		return "stats.om"
+	}
+
+	return "stats.json"
+}
+
+// CreateWorkloadHistogramExporter creates a exporter.Exporter based on the roachtest parameters
+func CreateWorkloadHistogramExporter(t test.Test, c cluster.Cluster) exporter.Exporter {
+	var metricsExporter exporter.Exporter
+	if t.ExportOpenmetrics() {
+		labels := clusterstats.GetOpenmetricsLabelMap(t, c, nil)
+		openMetricsExporter := &exporter.OpenMetricsExporter{}
+		_ = openMetricsExporter.SetLabels(&labels)
```

Comment:
nit: Maybe add a comment explaining we can ignore error checking here since we know the labels created by `clusterstats.GetOpenmetricsLabelMap` are already sanitized.